src/components/virtualRepeater/virtualRepeater.js

```diff
@@ -0,0 +1,452 @@
+ *
+ * Virtual repeat is a limited substitute for ng-repeat that allows renders only
+ * enough dom nodes to fill the container and recycling them as the user scrolls.
+ *
+ * @usage
+ * <hljs lang="html">
+ * <md-virtual-repeat-container md-horizonal>
+ *   <div md-virtual-repeat="i in items">Hello {{i}}!</div>
+ * </md-virtual-repeat-containr>
+ *
+ * <md-virtual-repeat-container md-horizonal>
+ *   <div md-virtual-repeat="i in items">Hello {{i}}!</div>
+ * </md-virtual-repeat-containr>
+ * </hljs>
+ *
+ * @param {boolean=} md-horizontal Whether the container should scroll horizontally
+ *     (defaults to scrolling vertically).
+ */
+function VirtualRepeatContainerDirective() {
+  return {
+    controller: VirtualRepeatContainerController,
+    replace: true,
+    require: 'virtualRepeatContainer',
+    restrict: 'E',
+    template: VirtualRepeatContainerTemplate
+  };
+}
+
+
+function VirtualRepeatContainerTemplate($element, $attrs) {
+  var innerHtml = $element[0].innerHTML;
+  $element[0].innerHTML = '';
+  // TODO: we probably don't need the container anymore.
+  return '<div class="md-virtual-repeat-container ' +
+      ($attrs.mdHorizontal ? 'md-horizontal' : 'md-vertical') + '">' +
+    '<div class="md-virtual-repeat-scroller">' +
+      '<div class="md-virtual-repeat-sizer"></div>' +
+      '<div class="md-virtual-repeat-offsetter">' +
+        innerHtml +
+      '</div>' +
+    '</div>' +
+  '</div>';
+}
+
+
+function VirtualRepeatContainerController($scope, $element, $attrs, $timeout, $window) {
+  this.$scope = $scope;
+  this.$element = $element;
+  this.$attrs = $attrs;
+  this.$timeout = $timeout;
+  this.$window = $window;
+
+  this.size = 0;
+  this.scrollSize = 0;
+  this.scrollOffset = 0;
+  this.repeater = null;
+
+  this.scroller = $element[0].getElementsByClassName('md-virtual-repeat-scroller')[0];
+  this.sizer = this.scroller.getElementsByClassName('md-virtual-repeat-sizer')[0];
+  this.offsetter = this.scroller.getElementsByClassName('md-virtual-repeat-offsetter')[0];
+
+  this.handleScroll_ = angular.bind(this, this.handleScroll_, !!$attrs.mdHorizontal);
+
+  $window.requestAnimationFrame(angular.bind(this, this.updateSize));
+  if ($attrs.ngHide) {
+    $scope.$watch($attrs.ngHide, angular.bind(this, function(hidden) {
+      if (!hidden) {
+        $window.requestAnimationFrame(angular.bind(this, this.updateSize));
+      }
+    }));
+  }
+}
+
+
+/** Called by the md-virtual-repeat inside of the container at startup. */
+VirtualRepeatContainerController.prototype.register = function(repeaterCtrl) {
+  this.repeater = repeaterCtrl;
+  
+  angular.element(this.scroller)
+      .on('scroll wheel touchmove touchend', this.handleScroll_);
+};
+
+
+/** @return {boolean} Whether the container is configured for horizontal scrolling. */
+VirtualRepeatContainerController.prototype.isHorizontal = function() {
+  return !!this.$attrs.mdHorizontal;
```

Comment:
Since `mdHorizontal` is a boolean attribute, this should always evaluate to false.
Something like `this.$attrs.hasOwnProperty('mdHorizontal')` should be used instead (here and elsewhere).